subseas_flex_fcst/maproom.py

```diff
@@ -94,6 +94,33 @@ def display_relevant_control(variable):
     return style_percentile, style_threshold
 
 
+@APP.callback(
+    Output("leadTime","options"),
+    Output("leadTime","value"),
+    Input("startDate","value"),
+)
+def targetStartOptions(startDate):
+    leadsValues = list(CONFIG["leads"].values())
+    leadsKeys = list(CONFIG["leads"])
+    startDate = pd.to_datetime(startDate)
+    optionsDict = {}
+    for idx, x in enumerate(leadsValues):
+        endDate = x+CONFIG["target_period_length"]-1 #used to calculate the target end date (targetEnd = lead + period length - 1)
```

Comment:
I would find simpler to read this if the date for the target start and the date for the target end were computed first once and for all (for each loop iteration). Then you run the tests to decide the format of the target start date. Note that the format of the target end date is always the same so need to include it in the test.
That would all save some lines.

Reply:
I agree. It took me several minutes to figure out what was intended here; I think it would have been easier if the calculation were separate from the formatting.

Incidentally, I also find the term "issue date" clearer than "start date" because it eliminates the ambiguity/overloading of "start" to mean either S (issue date) or S + L (start of target period).

Reply:
if we are it... I didn't want to load the boat (and I might be contradicting myself from the past here) but let's use the terms already in use (they probably what they are now for a good reason)

http://iridl.ldeo.columbia.edu/maproom/Global/ForecastsS2S/precip_flex_subx_weekly.html?S=0000%2023%20Sep%202022

I mean for the controls:
"Forecast Issued"
"Target Period"

Reply:
Okay. Should I be changing the variable names throughout the code to reflect this?

Reply:
@remicousin Let me know if you would like me to make a new issue for changing variable names to reflect this difference, or if you just wanted it applied to the controls names.

Reply:
Hopefully soon, we will have to generalize a good deal of the things here to work with seasonal forecast. That will be an occasion to rationalize names, and possibly easier to do or think of as we will have a new case to run this all through.

Reply:
Okay then I will leave that for now.